packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts

```diff
@@ -0,0 +1,174 @@
+
+    // generate CFN template
+    const template: Template = await this.synthesizeTemplates();
+
+    // check if the cfnModifiers are defined for the Auth Stack
+    if (this._cfnModifiers) {
+      this._cfnModifiers(template);
+    }
+
+    // save stack
+    if (this._command === CommandType.ADD || this._command === CommandType.UPDATE) {
+      await this.deployOverrideStacksToDisk({
+        templateStack: template,
+        rootFilePath: this._deploymentOptions!.rootFilePath,
+      });
+    }
+    return template;
+  }
+
+  private applyOverride = async () => {
+    if (this._command === CommandType.ADD || this._command === CommandType.UPDATE) {
+      const { overrideProps } = await import(this._overrideProps!.overrideFnPath);
+      if (typeof overrideProps === 'function' && overrideProps) {
+        // await buildOverrideDir(this._overrideProps!.overrideDir);
+        this._authTemplateObj = overrideProps(this._authTemplateObj as AmplifyAuthStackTemplate);
+      } else {
+        console.log('There is no override setup yet for Root Stack. To enable override : Run amplify override root');
+      }
+    }
+  };
+  /**
+   *
+   * @returns Object required to generate Stack using cdk
+   */
+  private getInput = async (): Promise<AmplifyAuthStackOptions> => {
+    return {
+      authStackFileName: this._resourceConfig.stackFileName,
+      authStackInputPayload: this._authInputState.getCliInputPayload(),
+      event: CommandType.ADD,
+    };
+  };
+
+  /**
+   * Generates Root stack Template
+   * @returns CFN Template
+   */
+  private generateAmplifyAuthStackTemplate = async () => {
+    this._authTemplateObj = new AmplifyAuthCognitoStack(this.app!, 'AmplifyAuthCongitoStack', { synthesizer: this._synthesizer });
+    if (this._authInputState._authInputPayload?.triggers) {
+      // get Transform for functions
+      generateNestedAuthTriggerTemplate(category, this._authInputState._authInputPayload);
+      //generateStacksForAuthTriggers(this._authInputState._authInputPayload);
+    }
+    if (this._authInputState._authInputPayload?.userPoolGroupList) {
+      //generateStackforUserPoolGroups(this._authInputState._authInputPayload);
+    }
+    if (this._authInputState._authInputPayload?.adminQueries) {
+      //generateStackForAdminQueries(this._authInputState._authInputPayload);
+    }
+  };
+  /**
+   *
+   * @returns return CFN templates sunthesized by app
+   */
+  private synthesizeTemplates = async (): Promise<Template> => {
+    this.app?.synth();
+    const templates = this._synthesizer.collectStacks();
+    return templates.get('AmplifyAuthCognitoStack')!;
+  };
+
+  private deployOverrideStacksToDisk = async (props: DeploymentOptions) => {
+    if (this._authStackOptions!.event === CommandType.ADD || this._authStackOptions!.event === CommandType.UPDATE) {
+      JSONUtilities.writeJson(props.rootFilePath, props.templateStack);
+    }
+  };
+
+  public getRootStack(): AmplifyAuthCognitoStack {
```

Comment:
Why do we need to have this root stack function inside auth category?

Reply:
we dont need it here , might be copy error from root stack